gulpfile.js

```diff
@@ -160,21 +102,95 @@ gulp.task('generate-grids-topics', () => {
                     "igSelector": grid.igSelector
                 }
             }))
-            .pipe(gulp.dest(DOCFX_ARTICLES + grid.igPath));
+            .pipe(dest(DOCFX_ARTICLES + grid.igPath));
     }
-});
-
-gulp.task('build', [
-    'styles',
-    'cleanup',
-    'post-processor-configs',
-    'generate-grids-topics',
-    'build-site'
-]);
-
-gulp.task('build-travis', [
-    'styles',
-    'cleanup',
-    'post-processor-configs',
-    'generate-grids-topics'
-]);
\ No newline at end of file
+     setTimeout( () => eventEmitter.emit("finish"), 3000);
+     return eventEmitter;
+}
+
+const  styles = () => {
+  return src(`${DOCFX_TEMPLATE}/styles/sass/main.scss`)
+        .pipe(sass().on('error', sass.logError))
+        .pipe(
+            autoprefixer({
+                overrideBrowserslist: ['last 2 versions'],
+                cascase: false
+            })
+        )
+        .pipe(dest(`${DOCFX_TEMPLATE}/styles/css`));
+};
+
+const buildSite = async() => {
+     await shell.task(`docfx build ${DOCFX_CONF}`)(done =>  done());
+}
+
+const watchFiles = () => {
+    
+    const excluded = [
+        `!${DOCFX_ARTICLES}/grid/**`,
+        `!${DOCFX_ARTICLES}/treegrid/**`,
+        `!${DOCFX_ARTICLES}/hierarchicalgrid/**`
+    ];
+
+    // All topics that are not auto-generated and are specific to the respective grid, should be here.
+    const included = [
+        `${DOCFX_ARTICLES}/grid/grid.md`,
+        `${DOCFX_ARTICLES}/grid/groupby.md`,
+        `${DOCFX_ARTICLES}/grid/paste_excel.md`,
+        `${DOCFX_ARTICLES}/treegrid/tree_grid.md`,
+        `${DOCFX_ARTICLES}/treegrid/aggregations.md`,
+        `${DOCFX_ARTICLES}/treegrid/load_on_demand.md`,
+        `${DOCFX_ARTICLES}/hierarchicalgrid/hierarchical_grid.md`,
+        `${DOCFX_ARTICLES}/hierarchicalgrid/load_on_demand.md`
+    ];
+
+    // watch([`${DOCFX_TEMPLATE}/**/*`, `!${DOCFX_TEMPLATE}/styles/css`], series(build));
+    watch([
+           `${DOCFX_TEMPLATE}/**/*`,
+           `!${DOCFX_TEMPLATE}/styles/css/main.css`,
+          ,`${DOCFX_PATH}/components/*.md`,
+           `${DOCFX_PATH}/general/**/*.md`,
+           `${DOCFX_PATH}/themes/*.md`,
+           `${DOCFX_ARTICLES}/**`].concat(excluded).concat(included),
+            series(build, browserSyncReload));
```

Comment:
In order to pass the delay: 
```suggestion
            {delay: 3000},
            series(build, browserSyncReload));
```